### PR TITLE
Update EEZ file ref to our own cache

### LINKED ISSUE #328: Wind offshore capacity factor rule fails
### What happened?

The wind offshore capacity factor rule fails, at all spatial resolutions.

### Version

1.2.0.dev

### Relevant log output

```shell
snakemake --profile profiles/conda --configfile="config/default.yaml" "build/models/national/timeseries/supply/capacityfactors-wind-offshore.csv"
Config file config/default.yaml is extended by additional config specified via the command line.
Building DAG of jobs...
Using shell: /bin/bash
Provided cores: 2
Rules claiming more threads will be scaled down.
Job stats:
job                          count    min threads    max threads
-------------------------  -------  -------------  -------------
capacity_factors_offshore        1              1              1
total                            1              1              1

Select jobs to execute...

[Tue Apr  9 11:54:22 2024]
Job 0: Generate capacityfactor time series disaggregated by location on national resolution for wind-offshore.
Reason: Missing output files: build/models/national/timeseries/supply/capacityfactors-wind-offshore.csv

Activating conda environment: ../envs/snakemake/euro-calliope/9498f8bccfbf440ed1c00845fc76b38a_
Activating conda environment: ../envs/snakemake/euro-calliope/9498f8bccfbf440ed1c00845fc76b38a_
/Users/adrienmellot/Documents/GitHub/envs/snakemake/euro-calliope/9498f8bccfbf440ed1c00845fc76b38a_/lib/python3.9/site-packages/geopandas/array.py:275: ShapelyDeprecationWarning: The array interface is deprecated and will no longer work in Shapely 2.0. Convert the '.coords' to a numpy array instead.
  return GeometryArray(vectorized.points_from_xy(x, y, z), crs=crs)
Traceback (most recent call last):
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/.snakemake/scripts/tmpwp_m_t5h.capacityfactors_offshore.py", line 84, in <module>
    capacityfactors(
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/.snakemake/scripts/tmpwp_m_t5h.capacityfactors_offshore.py", line 55, in capacityfactors
    capacityfactors_per_eez = area_weighted_time_series(
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/lib/eurocalliopelib/geo/spatiotemporal.py", line 33, in area_weighted_time_series
    data={
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/lib/eurocalliopelib/geo/spatiotemporal.py", line 34, in <dictcomp>
    shape_id: weighted_time_series(
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/lib/eurocalliopelib/geo/spatiotemporal.py", line 84, in weighted_time_series
    assert ds.weight.sum() >= gridcell_overlap_threshold, ds.weight.sum()
AssertionError: <xarray.DataArray 'weight' ()>
array(0.58280955)
Coordinates:
    shape_id  int64 64459
[Tue Apr  9 11:55:28 2024]
Error in rule capacity_factors_offshore:
    jobid: 0
    input: build/data/eez.geojson, build/data/national/shared-coast.csv, data/automatic/capacityfactors/wind-offshore-timeseries.nc
    output: build/models/national/timeseries/supply/capacityfactors-wind-offshore.csv
    conda-env: /Users/adrienmellot/Documents/GitHub/envs/snakemake/euro-calliope/9498f8bccfbf440ed1c00845fc76b38a_

Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-04-09T115421.386948.snakemake.log
```

### PR DESCRIPTION
**Pull request by brynpickering**
Fixes #328 

The issue is a newer version of EEZ is being downloaded, which we can't control for on the `https://geo.vliz.be/geoserver/MarineRegions/wfs` file server. Versioned releases _are_ available [here](https://www.marineregions.org/downloads.php), but require a user to tick a box and fill in some info about themselves. The data license allows us to just upload it elsewhere, which I've done by putting it on the [`euro-calliope-datasets` repo](https://github.com/calliope-project/euro-calliope-datasets/pull/5). Arguably not the best way to handle caching a working version of EEZ (in this case, EEZ V11) but it is possibly the quickest.

Tested on the full workflow and it works to produce offshore wind data. Change from the v1.1 pre-built CSV file (average of hourly new cf / hourly old cf per country):

> [!NOTE]
> EDIT: Using high res EEZ data, the result is a 1:1 mapping now. No change in data.

```
AUT         NaN
BEL    1.000087
BGR    0.999992
HRV    0.999954
CYP    0.999995
CZE         NaN
DNK    0.999905
EST    0.999875
FIN    0.999110
FRA    1.000038
DEU    1.000079
GRC    0.999975
HUN         NaN
IRL    1.000002
ITA    0.999969
LVA    1.000151
LTU    0.999934
LUX         NaN
NLD    1.000147
POL    1.000209
PRT    1.000019
ROU    0.999999
SVK         NaN
SVN    0.998619
ESP    0.999996
SWE    0.999313
GBR    1.000024
ALB    1.000064
BIH    1.000000
MKD         NaN
MNE    0.999882
NOR    0.999733
SRB         NaN
CHE         NaN
```

## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [ ] CHANGELOG updated
- [ ] Minimal workflow tests pass
- [ ] Tests added to cover contribution
- [ ] Documentation updated
- [ ] Configuration schema updated

### CODE REVIEW

**Comment by brynpickering** [comment]
I've updated to use zenodo. For now the data is in a [sandbox record](https://sandbox.zenodo.org/records/45135), with the plan to move it to [the main input record](https://zenodo.org/records/5797549) when we release v1.2.

File: CHANGELOG.md [hunk 1]
```diff
@@ -56,6 +56,7 @@
 ### Fixed (workflow)
 
 * **FIX** fixed optimisation tolerance of hydro power plants from xtol to xatol (#266).
+* **FIX** source of Exclusive Economic Zones (EEZ) to use a cache on [zenodo](https://sandbox.zenodo.org/records/45135) so that we can keep using v11 (#332).
```
Comment (timtroendle) [inline CHANGELOG.md:59]: Best add a "FIXME update to actual data store before release of Euro-Calliope"

File: config/default.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # For descriptions of the configuration values, see `./schema.yaml`.
 data-sources:
     biofuel-potentials-and-costs: https://cidportal.jrc.ec.europa.eu/ftp/jrc-opendata/ENSPRESO/ENSPRESO_BIOMASS.xlsx
-    eez: https://geo.vliz.be/geoserver/MarineRegions/wfs?service=WFS&version=1.0.0&request=GetFeature&typeNames=MarineRegions:eez&outputFormat=SHAPE-ZIP
+    eez: https://sandbox.zenodo.org/records/45135/files/eez_v11.gpkg.zip?download=1
```
Comment (timtroendle) [inline config/default.yaml:4]: Add the FIXME here too.

**Review by timtroendle — CHANGES_REQUESTED** [review]
Looks good. Please add a FIXME because we will easily forget.

**Comment by timtroendle** [comment]
Btw do you know how often Sandbox is wiped?

**Comment by brynpickering** [comment]
> Btw do you know how often Sandbox is wiped?

I don't. I couldn't find any info on it either. Oldest available record is Feb 3rd, so maybe ~2 months? Gives us a deadline...

**Comment by brynpickering** [comment]
> In the sandbox environment, you can try out all functions of Zenodo and the submitted data will be deleted from time to time. 

https://www.uzh.ch/blog/hbz/2018/11/14/how-to-submit-in-zenodo/?lang=en

**Comment by brynpickering** [comment]
Maybe we upload to zenodo as a unique record?

**Comment by timtroendle** [comment]
As discussed, I'd be fine with using sandbox. We only need to release within 2 months and there won't be any problems, haha.

**Pull request merged**